File: vben-admin/apps/web-antd/src/api/checkin.ts
import { requestClient } from '#/api/request';

export interface CheckinStatusResult {
  checked_in: boolean;
  reward_money?: number;
}

export interface CheckinResult {
  reward_money: number;
}

export interface CheckinRecord {
  uid: number;
  username: string;
  reward_money: number;
  addtime: string;
}

export interface CheckinStatsResult {
  total_users: number;
  total_reward: number;
  list: CheckinRecord[];
  total: number;
}

export function userCheckinApi() {
  return requestClient.post<CheckinResult>('/user/checkin');
}

export function userCheckinStatusApi() {
  return requestClient.get<CheckinStatusResult>('/user/checkin/status');
}

export function adminCheckinStatsApi(params: { date?: string; page?: number; limit?: number }) {
  return requestClient.get<CheckinStatsResult>('/admin/checkin/stats', { params });
}

File: go-api/internal/model/checkin.go
package model

type CheckinRecord struct {
	ID          int     `json:"id"`
	UID         int     `json:"uid"`
	Username    string  `json:"username"`
	RewardMoney float64 `json:"reward_money"`
	CheckinDate string  `json:"checkin_date"`
	AddTime     string  `json:"addtime"`
}

type CheckinDayStat struct {
	CheckinDate string  `json:"checkin_date"`
	TotalUsers  int     `json:"total_users"`
	TotalReward float64 `json:"total_reward"`
}
